### DISCUSSION TITLE
Read .TIFF can not find image width height (from EXIF)

### DISCUSSION THREAD

**Discussion by fred888651** [ImageMagick/ImageMagick#2518] [Help]
ImageMagick-7.0.9-Q16 - Windows 10 - PowerShell 5.1
(Photoshop CS6 - Lightroom CS6)

**magick identify -define tiff:exif=true -verbose .\IMG_5914.tif**

syntax works - gives several pages of information  (mostly photoshop edit history in great detail)
**_ISSUE:  cant find - width and height!
QUESTION: How to extract image heigth and widht from .TIF file (from EXIF)__**
Gives some non-terminating exceptions - see below

(Also tried:
magick identify -define tiff:exif=true -format "%w %h" .\IMG_5914.tif

Gives exceptions - wrong syntax it seems
)

**magick identify -define tiff:exif=true -verbose .\IMG_5914.tif**

Gives output - but some (non-terminating) exceptions
Several unkown field tags - is this image heigth and width?

"

magick : identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/1025.
At line:1 char:1
+ magick identify -define tiff:exif=true -verbose .\IMG_5913.tif
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (identify: Incom...FWarnings/1025.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
Location: 4,029 Inches, 3,979 Inches

Set Source Sampling Point of current application	To: Document Location
Source: layer â€œLayer 1â€
Location: 4,092 Inches, 3,942 Inches

identify: Unknown field with tag 42032 (0xa430) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42033 (0xa431) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42034 (0xa432) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42036 (0xa434) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.

"

**Comment by fmw42** [comment]
remove the -define  

`magick identify -format "%w %h" .\IMG_5914.tif`

also your define syntax is wrong. it should be

`-define tiff:exif-properties=true`  

not

`-define tiff:exif=true  `


See https://imagemagick.org/script/defines.php

**Reply by fred888651** [reply]
Thanks frmw42!

First suggestion works great for JPGs and somewhat ok with CR2 (Canon RAW files)

But not for TIFs:

**magick identify -format "%w %h" .\IMG_5914.tif**
magick : identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/1025.
At line:1 char:1
+ magick identify -format "%w %h" .\IMG_5914.tif
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (identify: Incom...FWarnings/1025.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
identify: Unknown field with tag 42032 (0xa430) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42033 (0xa431) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42034 (0xa432) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42036 (0xa434) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
2024 20242024 20242024 20242024 20242024 2024




Second suggestions returns:

**magick identify -define tiff:exif-properties=true .\IMG_5914.tif**

.\IMG_5914.tif[0] TIFF 2024x2024 2024x2024+0+0 8-bit sRGB 52.7986MiB 0.281u 0:00.275
.\IMG_5914.tif[1] TIFF 2024x2024 2024x2024+0+0 8-bit sRGB 52.7986MiB 0.031u 0:00.026
.\IMG_5914.tif[2] TIFF 2024x2024 2024x2024+0+0 8-bit sRGB 52.7986MiB 0.031u 0:00.025
.\IMG_5914.tif[3] TIFF 2024x2024 2024x2024+0+0 8-bit sRGB 52.7986MiB 0.031u 0:00.025
.\IMG_5914.tif[4] TIFF 2024x2024 2024x2024+0+0 8-bit sRGB 52.7986MiB 0.031u 0:00.025
magick : identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/1025.
At line:1 char:1
+ magick identify -define tiff:exif-properties=true .\IMG_5914.tif
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (identify: Incom...FWarnings/1025.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
identify: Unknown field with tag 42032 (0xa430) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42033 (0xa431) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42034 (0xa432) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.
identify: Unknown field with tag 42036 (0xa434) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/1025.

**Reply by fred888651** [reply]
Can IM read TIFFs with multiple layers today? Has this been solved?

Im a photographer and my TIFFs (photoshop CS6) usually have lots of layers, image layers and adjustment layers.
(IMG_5914.tif has 3 image layers and 5 adjustment layers = 8 layers)

https://www.imagemagick.org/discourse-server/viewtopic.php?t=30165

_Re: Problem opening Converted TIFF Images with Multiple Layers in PhotoShop after Conversion
Post by fmw42 » 2016-07-26T17:44:20+02:00

Your image has adjustment layers. Imagemagick cannot currently handle them properly as far as I know. It can only deal with simple layers._

**Reply by fmw42** [reply]
As far as I know, it cannot read TIFF with layers, but can read tiff with pages.  Many of your messages are warning about unknown custom tags. You can avoid those by adding -quiet right after magick or magick identify.

I suspect your errors are about layers in the tiff.  But I will defer to the IM developers for further comment.

**Reply by fred888651** [reply]
Ok!

Yes, its a lot of layers in my (Photoshop) TIFs
What is the difference between a layer and a page?

-quiet seems nice - at the moment I am ignoring exceptions from IM by other means 

(
PowerShell: $ErrorAction = "SilentlyContinue"

-ErrorAction SilentlyContinue & try/catch still produced red text - (cmdlet above was only alternative that were completely silent)

)

**Comment by snibgo** [comment]
@fred888651: You asked for:
```
-format "%w %h"
```
... and got the answer:
```
2024 20242024 20242024 20242024 20242024 2024
```
So the input has five images, each are 2024x2024. If you put /n at the end of your format, you will get new lines between each, so they are easier to read, like this:
```
2024 2024
2024 2024
2024 2024
2024 2024
2024 2024
```

**Reply by fred888651** [reply]
Thanks again!

magick identify '-format "%w %h"/n' .\IMG_5914.tif
magick identify '-format /n "%w %h"' .\IMG_5914.tif
magick identify -format  "%w %h /n" .\IMG_5914.tif

Returns syntax errors - a little diff for last one

$WH = magick.exe identify -format "%w %h"  .\IMG_5913.tif
Gives exceptions , but these I can swallow 
Returns
1177 11771177 11771177 11771177 1177

So seems like the way to go
Just need to cut out from that string $WH to 
get $W=1117 and $H=1177

Thanks!

**Reply by fmw42** [reply]
use \n, not /n

**Reply by fred888651** [reply]
Thanks! That solved another problem
A lot easier to pick out first element this way!

**Comment by snibgo** [comment]
Sorry, I mean backslash-n, \n, the usual C character for "newline".
```
-format "%w %h\n"
```

**Reply by fred888651** [reply]
Thanks!  Works great!

Solved another problem!

(`n is the newline character in PowerShell - why o why - we need a new world order to get rid of all these different syntax for the same old things :) )

**Comment by snibgo** [comment]
We can put anything we want in the format. This means we can do useful things like this (bash syntax):
```
$ declare $(magick.exe identify -format "MYW_%s=%w\nMYH_%s=%h\n" x.tiff)
```
My file x.tiff has two images, so the magick command writes text output like this:
```
MYW_0=267
MYH_0=233
MYW_1=70
MYH_1=46
```
Running this as a sub-command of "declare" creates four environment variables.
